src/stores/ParityStore.js

```diff
@@ -0,0 +1,141 @@
+// Copyright 2015-2018 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+//
+// SPDX-License-Identifier: MIT
+
+import Api from '@parity/api';
+import { action, observable } from 'mobx';
+import isElectron from 'is-electron';
+import light from '@parity/light.js';
+import store from 'store';
+
+const electron = isElectron() ? window.require('electron') : null;
+
+const LS_PREFIX = '__paritylight::';
+const LS_KEY = `${LS_PREFIX}secureToken`;
+
+export default class ParityStore {
+  @observable downloadProgress = 0;
+  @observable isApiConnected = false;
+  @observable isParityRunning = false;
+  @observable token = null;
+
+  constructor () {
+    // Retrieve token from localStorage
+    const token = store.get(LS_KEY);
+    if (token) {
+      this.setToken(token);
+    }
+
+    if (!electron) {
+      console.log(
+        'Not in Electron, ParityStore will only have limited capabilities.'
+      );
+      return;
+    }
+
+    const { ipcRenderer, remote } = electron;
+
+    // Check if isParityRunning
+    this.setIsParityRunning(!!remote.getGlobal('isParityRunning'));
+    // We also listen to future changes
+    ipcRenderer.on('parity-running', (_, isParityRunning) => {
+      this.setIsParityRunning(isParityRunning);
+    });
+
+    // Set download progress
+    ipcRenderer.on('parity-download-progress', (_, progress) => {
+      this.setDownloadProgress(progress);
+    });
+  }
+
+  connectToApi = () => {
+    let provider = 'ws://127.0.0.1:8546'; // Default provider to be used in localhost:3000
+    if (electron) {
+      const { remote } = electron;
+      const wsInterface = remote.getGlobal('wsInterface');
+      const wsPort = remote.getGlobal('wsPort');
+      if (wsInterface && wsPort) {
```

Comment:
Ah, just noticed this doesn't work if we only put as flag `--ws-port 1234`, will still output `ws://127.0.0.1:8546`